storage_service/ARCHITECTURE.md

```diff
@@ -177,20 +198,38 @@ Return Verification Result
 storage_service/
 ├── src/
 │   ├── main.rs              # HTTP server and API endpoints
-│   ├── hash.rs              # Hashing utilities
 │   └── merkle_tree/         # Merkle tree implementation
 │       ├── mod.rs           # Module exports
-│       ├── hasher.rs        # Hashing trait and implementations
-│       ├── proof.rs         # Proof structures
+│       ├── hasher.rs        # Hashing trait (delegates to atlas-common)
+│       ├── proof.rs         # Proof structures and MerkleProof trait
 │       └── tree.rs          # Core Merkle tree logic
 ```
 
+### Dependencies
+
+#### Core Dependencies
+- **atlas-common**: Standardized cryptographic operations and validation
+  - Features: `["hash", "validation"]`
+  - Provides: SHA256/384/512 hashing, manifest validation, hash verification
+- **actix-web**: HTTP server framework
+- **mongodb**: Database driver
+- **ring**: Ed25519 signatures
+- **serde**: Serialization/deserialization
+
+#### Atlas-Common Integration
+- **Hashing**: All SHA operations delegated to `atlas-common::hash`
+- **Validation**: Manifest ID and hash validation via `atlas-common::validation`
+- **Consistency**: Ensures uniform cryptographic behavior across Atlas framework
+- **Future-Proof**: Centralized updates to cryptographic functions
+
 ### Key Design Decisions
 
-1. **SHA384 over SHA256**: Provides additional security margin
-2. **In-Memory Merkle Tree**: Fast proof generation with MongoDB persistence
-3. **Flexible Content Formats**: Supports JSON, CBOR, and binary manifests
-4. **Synchronous Tree Updates**: Ensures consistency but may impact latency
+1. **SHA384 over SHA256**: Provides additional security margin (consistent with atlas-common default)
+2. **atlas-common Integration**: Eliminates code duplication and ensures consistency
+3. **In-Memory Merkle Tree**: Fast proof generation with MongoDB persistence
+4. **Flexible Content Formats**: Supports JSON, CBOR, and binary manifests
+5. **Synchronous Tree Updates**: Ensures consistency but may impact latency
+6. **Centralized Validation**: Uses atlas-common for all validation logic
```

Comment:
This may be misunderstood as there being a centralized validation service or so. Maybe something like "Unified Validation" would be less ambiguous.
```suggestion
6. **Unified Validation**: Uses atlas-common for all validation logic
```